Assets/Character/Scripts/Movement.cs

```diff
@@ -0,0 +1,31 @@
+﻿using UnityEngine;
+using System.Collections;
+
+public class Movement : MonoBehaviour
+{
+	private PlayerStateController _controller;
+    private Rigidbody2D _rb;
+
+	void Start()
+	{
+		_controller = this.GetComponent<PlayerStateController>();
+        _rb = this.GetComponent<Rigidbody2D>();
+	}
+
+    void FixedUpdate()
+    {
+        if (_controller.InState(Constants.STATE_MOVE_LEFT))

```

Comment:
maybe have getState() and do a switch on it. may be cleaner than this if/else structure in this file and the other file